clickhouse-admin/src/context.rs

```diff
@@ -3,84 +3,555 @@
+    },
+}
+
+async fn long_running_db_init_task(incoming: Receiver<DbInitRequest>) {
+    while let Ok(request) = incoming.recv_async().await {
+        match request {
+            DbInitRequest::DbInit {
+                clickhouse_address,
+                log,
+                replicated,
+                response,
+            } => {
+                let result =
+                    init_db(clickhouse_address, log.clone(), replicated).await;
+                if let Err(e) = response.send(result) {
+                    error!(
+                        log,
+                        "failed to send value from database initialization to channel: {e:?}"
+                    );
+                };
+            }
+        }
+    }
+}
+
+pub enum GenerateConfigRequest {
+    /// Generates a configuration file for a server or keeper node
+    GenerateConfig {
+        clickward: Clickward,
+        log: Logger,
+        node_settings: NodeSettings,
+        response: oneshot::Sender<Result<GenerateConfigResult, HttpError>>,
+    },
+}
+
+async fn long_running_generate_config_task(
+    incoming: Receiver<GenerateConfigRequest>,
+    generation_tx: watch::Sender<Option<Generation>>,
+) {
+    while let Ok(request) = incoming.recv_async().await {
+        match request {
+            GenerateConfigRequest::GenerateConfig {
+                clickward,
+                log,
+                node_settings,
+                response,
+            } => {
+                let result = generate_config_and_enable_svc(
+                    generation_tx.clone(),
+                    clickward,
+                    node_settings,
+                );
+                if let Err(e) = response.send(result) {
+                    error!(
+                        &log,
+                        "failed to send value from configuration generation to channel: {e:?}"
+                    );
+                };
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum NodeSettings {
+    Replica { settings: ServerConfigurableSettings, fmri: String },
+    Keeper { settings: KeeperConfigurableSettings, fmri: String },
+}
+
+impl NodeSettings {
+    fn fmri(self) -> String {
+        match self {
+            NodeSettings::Replica { fmri, .. } => fmri,
+            NodeSettings::Keeper { fmri, .. } => fmri,
+        }
+    }
+
+    fn generation(&self) -> Generation {
+        match self {
+            NodeSettings::Replica { settings, .. } => settings.generation(),
+            NodeSettings::Keeper { settings, .. } => settings.generation(),
+        }
+    }
+}
+
+pub fn generate_config_and_enable_svc(
+    generation_tx: watch::Sender<Option<Generation>>,
+    clickward: Clickward,
+    node_settings: NodeSettings,
+) -> Result<GenerateConfigResult, HttpError> {
+    let incoming_generation = node_settings.generation();
+    let generation_rx = generation_tx.subscribe();
+    let current_generation = *generation_rx.borrow();
```

Comment:
We don't need to subscribe here; the sending side also has a `.borrow()` method:

```suggestion
    let current_generation = *generation_tx.borrow();
```